backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/view/StylesFactory.java

```diff
@@ -65,32 +67,54 @@ public EdgeStyle createEdgeStyle(org.eclipse.sirius.web.view.EdgeStyle edgeStyle
     }
 
     public String getNodeType(NodeStyle nodeStyle) {
-        if (nodeStyle.getShape() == null || nodeStyle.getShape().isBlank()) {
-            return NodeType.NODE_RECTANGLE;
-        } else {
-            return NodeType.NODE_IMAGE;
+        String type = NodeType.NODE_RECTANGLE;
+        if (nodeStyle.isListMode()) {
+            type = NodeType.NODE_LIST;
+        } else if (nodeStyle.eContainer().eContainer() instanceof org.eclipse.sirius.web.view.NodeDescription
+                && ((org.eclipse.sirius.web.view.NodeDescription) nodeStyle.eContainer().eContainer()).getStyle().isListMode()) {
+            type = NodeType.NODE_LIST_ITEM;
+        } else if (nodeStyle.getShape() != null && !nodeStyle.getShape().isBlank()) {
+            type = NodeType.NODE_IMAGE;
         }
+        return type;
     }
 
     public INodeStyle createNodeStyle(NodeStyle nodeStyle, ICustomImagesService customImagesService) {
-        if (NodeType.NODE_IMAGE.equals(this.getNodeType(nodeStyle))) {
+        INodeStyle result = null;
+        switch (this.getNodeType(nodeStyle)) {
+        case NodeType.NODE_IMAGE:
             String shapeId = nodeStyle.getShape();
             String shapeFileName = customImagesService.findById(UUID.fromString(shapeId)).map(CustomImage::getFileName).orElse(DEFAULT_SHAPE_FILE);
+            result = ImageNodeStyle.newImageNodeStyle().scalingFactor(1).imageURL("/custom/" + shapeFileName).build(); //$NON-NLS-1$
```

Comment:
Put this under @formatter:off - @formatter:on

Reply:
Done.